File: algorithm/frames/examples/kickstart_2019AA.py
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function
from __future__ import unicode_literals

import sys
# import resource
# resource.setrlimit(resource.RLIMIT_STACK, [0x100000000, resource.RLIM_INFINITY])
# sys.setrecursionlimit(0x1000000)
# #when you need more memory for function recursion, uncomment above

if sys.version_info[0]==2:
    input=raw_input

def input_int_seq():
    s = input()
    return [int(i) for i in s.split()]

import math

MAXN = int(1e5+10)
MAXM = int(1e5+10)
INF = int(1e9+7)
llINF = 1e18+7
dir4 = [[0,1], [1,0], [0,-1], [-1,0]]
dir8 = [[0,1], [1,1], [1,0], [1,-1], [0,-1], [-1,-1], [-1,0], [-1,1]]
eps = 1e-10

N, M, T, Cases= 0, 0, INF, 0

sys.stdin = open('input.txt', 'r')
sys.stdout = open('output.txt', 'w')

T = int(input())
while(T>0):
    T-=1
    Cases+=1
    N, M = input_int_seq()

    # define your input procedure
    a = input_int_seq()
    a = sorted(a)+[0]
    pre_sum = 0
    ans = llINF
    for i in range(M):
        pre_sum += a[i]
    for i in range(M-1, N):
        if (M*a[i]-pre_sum<ans):
            ans = M*a[i]-pre_sum
        pre_sum += a[i+1]-a[i-M+1]
    
    # define your solution

    print('Case #%d:'%(Cases), ans)

File: algorithm/frames/examples/kickstart_2019AA.cpp
//#pragma comment(linker, "/STACK:1024000000,1024000000") //when you need more memory for function recursion, uncomment this
#include <bits/stdc++.h>
using namespace std;

const int MAXN = 1e5+10;
const int MAXM = 1e5+10;
const int INF = 1e9+7;
const long long llINF = 1e18+7;
const int dir4[4][2]={{0,1}, {1,0}, {0,-1}, {-1,0}};
const int dir8[8][2]={{0,1}, {1,1}, {1,0}, {1,-1}, {0,-1}, {-1,-1}, {-1,0}, {-1,1}};
const double eps = 1e-10;

long long a[MAXN];

long long N,M,T = INF,Cases = 0;

int main(){
    // ios_base::sync_with_stdio(0); // uncomment this if you need really fast input&output with cin&cout
    //****************COMMENT THIS*******************//
    // freopen("input.txt", "r", stdin); //work with stdin from input.txt
    // freopen("output.txt", "w", stdout); //work with stdout from output.txt
    //****************WHEN SUBMMIT*******************//

    // scanf("%d",&T);
    cin >> T;
    // while(T-- && ~(scanf("%d%d",&N,&M)) && Cases++){
    while(T-- && (cin >> N >> M) && ++Cases){

        // define your input procedure
        for (int i = 0;i < N;i++) cin >> a[i];
        
        // define your solution
        sort(a,a+N);
    	long long pre_sum = 0;
    	long long ans = llINF;
    	for (int i = 0;i < M;i++) pre_sum += a[i];
    	for (int i = M-1;i < N; i++){
    		if (M*a[i]-pre_sum<ans) ans = M*a[i]-pre_sum;
    		pre_sum += a[i+1]-a[i-M+1];
    	}
        
        // printf("Case #%d: ", Cases);
        cout << "Case #" << Cases << ": ";
        
        //define your output procedure
        // printf("%d\n", Ans);
        cout << ans << endl;

    }
    return 0;
}
